crates/api/src/externals.rs

```diff
@@ -34,7 +34,7 @@ impl Extern {
     /// Returns the underlying `Func`, if this external is a function.
     ///
     /// Returns `None` if this is not a function.
-    pub fn func(&self) -> Option<&Func> {
+    pub fn into_func(self) -> Option<Func> {
```

Comment:
Could the old methods be preserved since they can be useful in their own right too?

Reply:
I'm not opposed, I just haven't found cases where they're useful. For the "I don't know what this is" or "This might be one of a few different things" use cases, `Extern` is just an `enum` and you can plain old `match` or `if let` on it, which read better anyway.

With the new `Instance::get_func` accessors, even `Extern::into_func` is a little dubious, because you can use `match` and `if let` to do that too, but they are handy when you want to report different errors in cases like this:

```rust
    instance
        .get_export("_start")
        .context("expected a _start export")?
        .into_func()
        .context("expected export to be a func")?
        .call(&[])?;
```